### PR TITLE
Updating wdb_agent.c methods with JSON parsing functions in wazuhdb_op.c

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#5638 #5639 #5654|

## Description

In this PR, the use of the method `wdbc_query_parse_json()` allows a more clear and concise implementation, preventing failures and resulting in a more clear source code.

It also fixes an issue related to the use of `sqlite_snprintf()` method in **wdb_find_agent()** function by replacing it by `snprintf`.

The registration test in Jenkins were failing because of a bug in a method inside **wdb_agent.c**. This is also fixed.
## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [x] Source installation
- [ ] Review logs syntax and correct language

### CODE REVIEW

File: src/wazuh_db/wdb_agent.c [hunk 1]
```diff
@@ -409,178 +359,108 @@ int wdb_remove_agent_db(int id, const char * name) {
...
-    cJSON *elem = NULL;
-    cJSON *json_name = NULL;
+    cJSON *json_offset = NULL;
+    char * column = NULL;
 
     switch (type) {
     case WDB_SYSCHECK:
         sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_SELECT_FIM_OFFSET], id_agent);
-        is_FIM = TRUE;
+        os_strdup("fim_offset",column);
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_agent.c:435]: `column` should be freed at the end of the method.

File: src/wazuh_db/wdb_agent.c [hunk 2]
```diff
@@ -611,61 +491,42 @@ int wdb_set_agent_offset(int id_agent, int type, long offset) {
...
 
-    if(strcmp(wdboutput,"ok") == 0 && strcmp(json_string,"[]") != 0 && strcmp(json_string,"[{}]") != 0 ){
-        root = cJSON_Parse(json_string);    
-        elem = cJSON_GetArrayItem(root, 0);
-        json_name = cJSON_GetObjectItem(elem, "status");
-        status = json_name->valuestring;
-        result = !strcmp(status, "empty") ? WDB_AGENT_EMPTY : !strcmp(status, "pending") ? WDB_AGENT_PENDING : WDB_AGENT_UPDATED;
+    json_status = cJSON_GetObjectItem(cJSON_GetArrayItem(root, 0),"status");
+    if(json_status){
+        os_strdup(json_status -> valuestring, status);
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_agent.c:513]: DITTO

File: src/wazuh_db/wdb_agent.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c:435]: Done!

File: src/wazuh_db/wdb_agent.c (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c:513]: Fixed!

File: src/wazuh_db/wdb_agent.c [hunk 3]
```diff
@@ -409,169 +359,102 @@ int wdb_remove_agent_db(int id, const char * name) {
...
 
-    if(json_string = wstr_chr(wdboutput, ' '), json_string ){
-        *json_string='\0';
-        json_string++;
-
-    } else{
-        mdebug1("SQLite result has no space. Query: %s",wdbquery);
-        os_free(array);
+    if (!root) {
+        merror("Error querying Wazuh DB to get the all agents.");
```
Comment (palaciosjeremias) [inline src/wazuh_db/wdb_agent.c:374]: Typo: Error querying Wazuh DB to get all agents.

File: src/wazuh_db/wdb_agent.c [hunk 4]
```diff
@@ -914,61 +753,36 @@ int wdb_delete_agent_belongs(int id_agent) {
...
+    os_calloc(n+1, sizeof(char *),array);        
 
-    } else{
-        mdebug1("SQLite Query failed: %s", wdbquery);
-        os_free(array);
-        return OS_INVALID;
+    for (i = 0; i < n; i++) {
+        elem = cJSON_GetArrayItem(root, i);
+        name = cJSON_GetObjectItem(elem, "name");
+        os_strdup(name->valuestring,array[i]);
```
Comment (palaciosjeremias) [inline src/wazuh_db/wdb_agent.c:786]: Can cJSON_GetObjectItem return NULL value? Should we check it?

File: src/wazuh_db/wdb_agent.c (hunk 3 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c:374]: Done!

File: src/wazuh_db/wdb_agent.c [hunk 5]
```diff
@@ -409,178 +359,108 @@ int wdb_remove_agent_db(int id, const char * name) {
...
 
-    if(json_string = wstr_chr(wdboutput, ' '), json_string ){
-        *json_string='\0';
-        json_string++;
-
-    } else{
-        mdebug1("SQLite result has no space. Query: %s",wdbquery);
-        os_free(array);
+    if (!root) {
+        merror("Error querying Wazuh DB to get the all agents.");
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c:374]: Removing "the"

File: src/wazuh_db/wdb_agent.c [hunk 6]
```diff
@@ -409,178 +359,108 @@ int wdb_remove_agent_db(int id, const char * name) {
...
-        root = cJSON_Parse(json_string);    
-        elem = cJSON_GetArrayItem(root, 0);
-        json_name = cJSON_GetObjectItem(elem, "id");
-        output = json_name->valueint;
-    } else{
-        mdebug1("SQLite Query failed: %s", wdbquery);
-        return OS_INVALID;
+    json_id = cJSON_GetObjectItem(cJSON_GetArrayItem(root, 0),"id");
+    if(json_id){
+        output = json_id -> valueint;
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c:416]: Done!

File: src/wazuh_db/wdb_agent.c [hunk 7]
```diff
@@ -409,178 +359,108 @@ int wdb_remove_agent_db(int id, const char * name) {
...
-            }
-        array[i] = -1;
-
-    } else{
-        mdebug1("SQLite Query failed: %s", wdbquery);
-        os_free(array);
-        return NULL;
+    for (i = 0; i < n; i++) {
+        json_id = cJSON_GetObjectItem(cJSON_GetArrayItem(root, i),"id");
+        if(json_id){
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c]: This cases are considered in the last fix

File: src/wazuh_db/wdb_agent.c [hunk 8]
```diff
@@ -409,178 +359,108 @@ int wdb_remove_agent_db(int id, const char * name) {
...
     switch (type) {
     case WDB_SYSCHECK:
         sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_SELECT_FIM_OFFSET], id_agent);
-        is_FIM = TRUE;
+        os_strdup("fim_offset",column);
         break;
     case WDB_SYSCHECK_REGISTRY:
         sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_SELECT_REG_OFFSET],id_agent);
-        is_FIM = FALSE;
+        os_strdup("reg_offset",column);
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c]: Heap allocation removed

File: src/wazuh_db/wdb_agent.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c]: Fixed!

File: src/wazuh_db/wdb_agent.c (hunk 4 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_agent.c]: The new implementation considers these cases according to the documentation
[cJSON](https://github.com/DaveGamble/cJSON)

**Pull request merged**